app/vmctl/testdata/servers_integration_test/remote_write_server.go

```diff
@@ -54,6 +56,22 @@ func NewRemoteWriteServer(t *testing.T) *RemoteWriteServer {
 	mux.Handle("/api/v1/label/__name__/values", rws.valuesHandler())
 	mux.Handle("/api/v1/export/native", rws.exportNativeHandler())
 	mux.Handle("/api/v1/import/native", rws.importNativeHandler(t))
+
+	if isCluster {
+		tenants := getTenants()
+		mux.Handle("/admin/tenants", rws.tenantsHandler(tenants))
+
+		for tenant := range tenants {
+			val := strconv.Itoa(tenant)
+			exportAPIPerTenant := fmt.Sprintf("/select/%s/prometheus/api/v1/export/native", val)
+			importAPIPerTenant := fmt.Sprintf("/insert/%s/prometheus/api/v1/import/native", val)
+			valuesAPIPerTenant := fmt.Sprintf("/select/%s/prometheus/api/v1/label/__name__/values", val)
+			mux.Handle(exportAPIPerTenant, rws.exportNativeHandler())
+			mux.Handle(importAPIPerTenant, rws.importNativeHandler(t))
+			mux.Handle(valuesAPIPerTenant, rws.valuesHandler())
+		}
+	}
+
```

Comment:
All the rest handlers above (*/api/v1/.*) should respect the tenant ID as well. 
Right now it is confusing since cluster mode works only for `/export/import` handlers. 
And `/select/%s/prometheus/api/v1/label/__name__/values` is conflicting with `/api/v1/label/__name__/values`.

I'd suggest spearating these servers to `NewRemoteWriteServer ` and `NewRemoteWriteCluster` with a unique set of handlers.